pkg/envoy/xds_server.go

```diff
@@ -217,24 +217,19 @@ func newXDSServer(envoySocketDir string, ipCache IPCacheEventSource) (*xdsServer
 
 // start configures and starts the xDS GRPC server.
 func (s *xdsServer) start() error {
-	// Remove/Unlink the old unix domain socket, if any.
-	_ = os.Remove(s.socketPath)
-
-	socketListener, err := net.ListenUnix("unix", &net.UnixAddr{Name: s.socketPath, Net: "unix"})
+	socketListener, err := s.newSocketListener()
 	if err != nil {
-		return fmt.Errorf("failed to open xDS listen socket at %s: %w", s.socketPath, err)
+		return fmt.Errorf("failed to create socket listener: %w", err)
 	}
 
-	// Make the socket accessible by owner and group only. Group access is needed for Istio
-	// sidecar proxies.
-	if err = os.Chmod(s.socketPath, 0660); err != nil {
-		return fmt.Errorf("failed to change mode of xDS listen socket at %s: %w", s.socketPath, err)
-	}
-	// Change the group to ProxyGID allowing access from any process from that group.
-	if err = os.Chown(s.socketPath, -1, option.Config.ProxyGID); err != nil {
-		log.WithError(err).Warningf("Envoy: Failed to change the group of xDS listen socket at %s, sidecar proxies may not work", s.socketPath)
-	}
+	resourceConfig := s.initializeXdsConfigs()
 
+	s.stopFunc = startXDSGRPCServer(socketListener, resourceConfig, 5*time.Second)
```

Comment:
I know it's the old code, but if it's not too much trouble, lift the `5*time.Second` into a constant and document what it's about. Also wondering how resilient this is.. what happens if the timeout is hit? How do we recover from it?

Reply:
good hint - which should lead to deletion of some code - because eventually this duration is used to create a context with timeout... which isn't used at all :) 

will open a separate PR if this is ok for you

Reply:
filed the PR: https://github.com/cilium/cilium/pull/26747